app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewSliderViewModel.kt

```diff
@@ -17,14 +17,60 @@
  */
 package com.infomaniak.drive.ui.fileList.preview
 
+import android.content.Context
+import android.content.Intent
+import androidx.core.content.FileProvider
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.infomaniak.drive.R
 import com.infomaniak.drive.data.models.File
 import com.infomaniak.drive.data.models.UserDrive
+import com.infomaniak.drive.ui.fileList.BaseDownloadProgressDialog.DownloadAction
+import com.infomaniak.drive.utils.Utils.openWith
+import com.infomaniak.drive.utils.printPdf
+import com.infomaniak.drive.utils.saveToKDrive
+import com.infomaniak.drive.utils.shareFile
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
 class PreviewSliderViewModel : ViewModel() {
+
+    val downloadProgressLiveData = MutableLiveData(0)
     val pdfIsDownloading = MutableLiveData<Boolean>()
     var currentPreview: File? = null
     var userDrive = UserDrive()
     var shareLinkUuid = ""
+
+    fun executeDownloadAction(
+        activityContext: Context,
```

Comment:
Maybe, instead of passing the activityContext in `executeDownloadAction`, we can use an `AndroidViewModel` ?

Reply:
We can't here, because this function needs an activity context, whereas the AndroidViewModel will give an application context